# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/ServoStyleSet.h
# Commit: 0347d8673dee
# Full Hash: 0347d8673deea522ae15120fe89e2552bf822a6d
# Author: Emily McDonough <emcdonough@mozilla.com>
# Date: 2022-12-03 09:24:59
# Regressor Bug: 1800546
# File Overlap Count: 1
# Description:
#   Bug 1800546 Part 1 - Use the style given the first page name for setting default orientation when printing r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D162058
# ==============================================================================

diff -r 0cda34ba7391 -r 0347d8673dee layout/style/ServoStyleSet.h
--- a/layout/style/ServoStyleSet.h	Sat Dec 03 01:15:55 2022 +0200
+++ b/layout/style/ServoStyleSet.h	Fri Dec 02 23:22:50 2022 +0000
@@ -253,12 +253,13 @@
   size_t SheetCount(Origin) const;
   StyleSheet* SheetAt(Origin, size_t aIndex) const;
 
-  // Gets the default orientation of unnamed CSS pages.
+  // Gets the default orientation of CSS pages.
   // This will return portrait or landscape both for a landscape/portrait
   // value to page-size, as well as for an explicit size or paper name which
   // is not square.
   // If the value is auto or square, then returns nothing.
-  Maybe<StylePageSizeOrientation> GetDefaultPageSizeOrientation();
+  Maybe<StylePageSizeOrientation> GetDefaultPageSizeOrientation(
+      const nsAtom* aFirstPageName);
 
   void AppendAllNonDocumentAuthorSheets(nsTArray<StyleSheet*>& aArray) const;
 
